### ISSUE TITLE
Cosine similarity

### ISSUE THREAD

**Issue by sudipansaha** [zaeemzadeh/OOD#1]
Can you please point out exactly where the cosine similarity is defined in the code?

**Comment by zaeemzadeh** [comment]
Hi,

In L93-94 of the `wideresnet_UoS.py`:
```python
out = F.normalize(out_feat, dim=1, p=2)
out = torch.abs(self.fc(out))
```
The feature vector is normalized and multiplied by the weight matrix. Note that the rows of the weight matrix are also normal. So each element in the `out` vector is the absolute value of the inner product of two normalized vectors, which is the absolute value of their cosine similarity.

**Comment by sudipansaha** [comment]
Thank you for the clarification.
